### PR TITLE
Fix deadlock on force reset if all instances are down

### PR DESCRIPTION
**Pull request by antoniofilipovic**
This diff fixes deadlock in force if all instances are down when an instance wants to be registered

### CODE REVIEW

**Comment by antoniofilipovic** [comment]
---
__*Leave above in PR description, copy the below into a comment*__
___


### Tracking
- [ ] **[Link to Epic/Issue]**


### Standard development
- [ ] Update unit/E2E tests
- [ ] Compare the [benchmarking results](https://bench-graph.memgraph.com/) between the master branch this branch


### CI Testing Labels
- [ ] Select the appropriate CI test labels _(CI -build=**build-name** -test=**test-suite**)_


### Documentation checklist
- [ ] Add the documentation label
- [ ] Add the bug / feature label
- [ ] Add the milestone for which this feature is intended
    - If not known, set for a later milestone
- [ ] Write a release note, including added/changed clauses
    - **[ Release note text ]**
- [ ] **[ Documentation PR link memgraph/documentation#XXXX ]**
    - [ ] Is back linked to this development PR
- [ ] **[ Tag someone from docs team ]**

**Comment by antoniofilipovic** [comment]
@kgolubic tagging you here, docs will be written soon

File: src/coordination/coordinator_instance.cpp [hunk 1]
```diff
@@ -323,6 +323,15 @@ auto CoordinatorInstance::ForceResetCluster_() -> ForceResetClusterStateStatus {
...
 
+  // There is nothing to do if we don't have any instances
+  if (instances.empty()) {
+    if (!raft_state_->AppendCloseLock()) {
+      spdlog::error("Aborting force reset as we failed to close lock on action.");
+      return ForceResetClusterStateStatus::FAILED_TO_CLOSE_LOCK;
+    }
+    return ForceResetClusterStateStatus::SUCCESS;
+  }
+
```
Comment (as51340) [inline src/coordination/coordinator_instance.cpp:334]: Can we move this at the function's start so we don't do any non-necessary work if there are no instances?

Comment (antoniofilipovic) [reply src/coordination/coordinator_instance.cpp:334]: Actually, it should be here as we need to check if we added instance to repl_instances_ and stop checks there, and also there is lock opening and other things needed which are before, so this is correct spot in my opinion

**Review by as51340 — APPROVED** [review]

**Pull request merged**